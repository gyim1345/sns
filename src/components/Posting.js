import React, { useState } from "react";
import PropTypes from "prop-types";
import { Link, Route } from "react-router-dom";
import Comment from "./Comment";
import Remove from "./Remove";
import Edit from "./Edit";
import userStorage from "../stores/userStore";
import toTop from "./toTop";
import Like from "./Like";

function Posting({
  posting,
  comments,
  globalState,
  setGlobalState,
  addComment,
  onChangeComment,
  sizeOfPicture,
  userOfActivePage,
  setUserOfActivePage,
  currentUser,
  commentAPI
}) {
const [input] = useState([]);



const changeUser = () => {
  setUserOfActivePage(posting.userName);
};

  return (
    <div>
      <h1>
        <Link
          to={`/${posting.userName}`}
          onClick={() => {
            changeUser();
            toTop();
          }}
        >
          <img src={userStorage.getUserImage(posting.userName)} alt="" width={50} />
          {posting.userName}
        </Link>
      </h1>
      <Link to={`/${userOfActivePage}/posting/${posting.id}`} onClick={toTop}>
        <img src={posting.imageUrl} alt="" width={sizeOfPicture} />
        <li>
          [Title]:
          {posting.title}
        </li>
  
        {/* [Id]:
        {posting.id}
        Image: */}
      </Link>
      <Like 
        posting={posting}
        currentUser={currentUser}
        setGlobalState={setGlobalState}
      />
      <Route exact path={`/${userOfActivePage}/posting/${posting.id}`}>
        <Edit
          posting={posting}
          globalState={globalState}
          setGlobalState={setGlobalState}
          userOfActivePage={userOfActivePage}
          currentUser={currentUser}
        />
        <Remove
          posting={posting}
          globalState={globalState}
          setGlobalState={setGlobalState}
          userOfActivePage={userOfActivePage}
          currentUser={currentUser}
        />
        <Comment
          posting={posting}
          comments={comments}
          globalState={globalState}
          setGlobalState={setGlobalState}
          userOfActivePage={userOfActivePage}
          currentUser={currentUser}
          commentAPI={commentAPI}
        />
        <input
          value={input[posting.id]}
          onChange={e => onChangeComment(e, posting.id)}
        />
        <button
          type="button"
          onClick={e => addComment(e, posting.id)}
          id="buttonAddComment"
        >
          AddComment
        </button>
      </Route>
    </div>
  );
}

Posting.propTypes = {
  userOfActivePage: PropTypes.string,
  setUserOfActivePage: PropTypes.func,
  currentUser: PropTypes.string,
  sizeOfPicture: PropTypes.string,
  posting: PropTypes.exact({
    id: PropTypes.number,
    title: PropTypes.string,
    imageUrl: PropTypes.string,
    userName: PropTypes.string,
    like: PropTypes.arrayOf(PropTypes.string)
  }),
  comments: PropTypes.arrayOf(PropTypes.object),
  globalState: PropTypes.oneOfType([PropTypes.number, PropTypes.array]),
  setGlobalState: PropTypes.elementType,
  addComment: PropTypes.elementType,
  onChangeComment: PropTypes.elementType
};

Posting.defaultProps = {
  userOfActivePage: "",
  setUserOfActivePage: "",
  currentUser: "",
  sizeOfPicture: 0,
  posting: [],
  comments: {},
  globalState: 0,
  setGlobalState: 0,
  addComment: "",
  onChangeComment: ""
};

export default Posting;
