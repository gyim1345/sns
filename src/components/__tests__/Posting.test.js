import React from 'react';
import { mount } from 'enzyme';
import Posting from '../Posting';
import {
  post,
  posts,
  comments,
  currentUser,
  sizeOfPicture,
  currentUserName
} from '../../Variables';
import { BrowserRouter as Router } from 'react-router-dom';

describe('<Posting />', () => {
  it('renders <h1> and <div>', () => {
    const wrapper = shallow(
      <Router key="0">
        <Posting
          posting={post}
          postingALl={posts}
          commentAPI={comments}
          currentUser={currentUser}
          userOfActivePage={currentUser}
          sizeOfPicture={sizeOfPicture}
        />
      </Router>
    );
    expect(wrapper.props().children.props.posting).toBe(post);
    expect(wrapper.props().children.props.postingALl).toBe(posts);
    expect(wrapper.props().children.props.commentAPI).toBe(comments);
    expect(wrapper.props().children.props.currentUser).toBe(currentUser);
    expect(wrapper.props().children.props.userOfActivePage).toBe(currentUser);
    expect(wrapper.props().children.props.sizeOfPicture).toBe(sizeOfPicture);
    expect(wrapper).toMatchSnapshot();
  });
});
