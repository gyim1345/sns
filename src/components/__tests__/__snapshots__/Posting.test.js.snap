// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posting /> renders <h1> and <div> 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Posting
    commentAPI={
      Array [
        Object {
          "_id": "5e90119435a2d94f9e08e35a",
          "id": 1,
          "like": Array [
            "noone@gmail.com",
          ],
          "postLId": 1,
          "replyToCommentId": null,
          "title": "title1",
          "userName": "gibong@gmail.com",
        },
      ]
    }
    currentUser="gibong@gmail.com"
    posting={
      Object {
        "_id": "5e8ec52d9bb4913b5a844423",
        "id": 1,
        "imageUrl": "http://localhost:3000/static/images/1.jpg",
        "like": Array [
          "gibong@gmail.com",
        ],
        "tag": Array [
          "landscape",
        ],
        "title": "posting with id 1",
        "userName": "gibong@gmail.com",
      }
    }
    postingALl={
      Array [
        Object {
          "_id": "5e8ec52d9bb4913b5a844423",
          "id": 1,
          "imageUrl": "http://localhost:3000/static/images/1.jpg",
          "like": Array [
            "gibong@gmail.com",
          ],
          "tag": Array [
            "landscape",
          ],
          "title": "posting with id 1",
          "userName": "gibong@gmail.com",
        },
      ]
    }
    sizeOfPicture={
      Object {
        "height": "10px",
        "width": "10px",
      }
    }
    userOfActivePage="gibong@gmail.com"
  />
</Router>
`;
